tutorials/09-variational-inference/09_variational-inference.jmd

```diff
@@ -273,28 +268,38 @@ end
 
 # A handy helper function to rescale our dataset.
 function standardize(x)
-    return (x .- mean(x; dims=1)) ./ std(x; dims=1), x
+    return (x .- mean(x; dims=1)) ./ std(x; dims=1)
+end
+
+function standardize(x, orig)
+    return (x .- mean(orig; dims=1)) ./ std(orig; dims=1)
 end
 
 # Another helper function to unstandardize our datasets.
 function unstandardize(x, orig)
-    return (x .+ mean(orig; dims=1)) .* std(orig; dims=1)
+    return x .* std(orig; dims=1) .+ mean(orig; dims=1)
+end
+
+function unstandardize(x, mean_train, std_train)
+    return x .* std_train .+ mean_train
 end
 ```
 
 ```julia
 # Remove the model column.
 select!(data, Not(:Model))
 
-# Standardize our dataset.
-(std_data, data_arr) = standardize(Matrix(data))
-
 # Split our dataset 70%/30% into training/test sets.
-train, test = split_data(std_data, 0.7)
+train, test = split_data(data, 0.7)
+train_unstandardized = copy(train) #  save a copy of the train data for later unstandardize use; alternatively, you can just save the statistics (mean and var) of the traiing data.
+
+# Standardize both datasets.
+std_train = standardize(Matrix(train))
+std_test = standardize(Matrix(test), Matrix(train_unstandardized))
```

Comment:
Probably would be simpler to use existing functionality in some package but with the functions in the tutorial you could do
```suggestion

# Standardize both datasets.
std_train = standardize(Matrix(train))
std_test = standardize(Matrix(test), Matrix(train))
```
No need to copy anything since `train` is not updated in-place.

Reply:
@devmotion Actually `train` is over-written later as a Matrix in Line 311:
```
train = Matrix(train_cut[:, remove_names]);
test = Matrix(test_cut[:, remove_names]);
```
This is from the original version and I therefore reverted the use of `train_unstandardized` here.